lib/jmap/core/sort/comparator.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:equatable/equatable.dart';
+import 'package:jmap_dart_client/jmap/core/sort/collation_identifier.dart';
+import 'package:jmap_dart_client/jmap/core/sort/property.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+abstract class Comparator with EquatableMixin, OptionalIsAscending, OptionalCollation {
+  final Property property;
+
+  Comparator(this.property);
+
+  Map<String, dynamic> toJson();
+}
+
+mixin OptionalIsAscending {
+  @JsonKey(includeIfNull: false)
+  bool? isAscending;
+
+  void setIsAscending(bool value) {
+    isAscending = value;
+  }
+}
+
+mixin OptionalCollation {
+  @JsonKey(includeIfNull: false)
+  CollationIdentifier? collation;
```

Comment:
please also update to `CoreCapability` also

Reply:
You should re-read the spec. It is not Capability. It is an identifier of Collation

Reply:
in `CoreCapability` we have a `Collation` also, please strong type that like you. (currently it is a string)

Reply:
oki